src/processor/Package.ts

```diff
@@ -330,4 +331,63 @@ export class Package {
       pullAt(sd.rules, rulesToRemove);
     });
   }
+
+  private removeDateRules(): void {
+    let allDatesMatch = true;
+    let date: string;
+    [...this.profiles, ...this.extensions, ...this.valueSets, ...this.codeSystems].forEach(sd => {
+      for (const rule of sd.rules) {
+        if (!allDatesMatch) break; // If any date hasn't matched the others, stop looking at other rules
+        if (
+          rule instanceof ExportableCaretValueRule &&
+          rule.caretPath === 'date' &&
+          rule.path === ''
+        ) {
+          const dateValue = rule.value as string; // If the rule assigns a date, the value will be a string.
+          if (!date) date = dateValue; // Set the date to match
+          if (date !== dateValue) {
+            allDatesMatch = false;
+          }
+        }
+      }
+    });
+
+    // If all dates are the same, are defined to the second, and are in GMT,
+    // we can assume they were set by the IG Publisher and can be safely removed.
+    const dateTimeRegex = /^([0-9]([0-9]([0-9][1-9]|[1-9]0)|[1-9]00)|[1-9]000)(-(0[1-9]|1[0-2])(-(0[1-9]|[1-2][0-9]|3[0-1])(T([01][0-9]|2[0-3]):[0-5][0-9]:([0-5][0-9]|60)(\.[0-9]+)?(Z|(\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00)))?)?)?$/;
+    const dateTimeMatch = date?.match(dateTimeRegex);
+    const usesGMT = dateTimeMatch?.[12] === '+00:00'; // dateTimeMatch[12] will hold timezone info if present
+    if (allDatesMatch && usesGMT) {
+      const DEFAULT_DATE = new ExportableCaretValueRule('');
+      DEFAULT_DATE.caretPath = 'date';
+      DEFAULT_DATE.value = date;
+      [...this.profiles, ...this.extensions].forEach(sd => {
+        const rulesToRemove: number[] = [];
+        sd.rules.forEach((rule, i) => {
+          if (isEqual(rule, DEFAULT_DATE)) {
+            rulesToRemove.push(i);
+          }
+        });
+        pullAt(sd.rules, rulesToRemove);
+      });
+      this.valueSets.forEach(vs => {
+        const rulesToRemove: number[] = [];
+        vs.rules.forEach((rule, i) => {
+          if (isEqual(rule, DEFAULT_DATE)) {
+            rulesToRemove.push(i);
+          }
+        });
+        pullAt(vs.rules, rulesToRemove);
+      });
+      this.codeSystems.forEach(cs => {
+        const rulesToRemove: number[] = [];
+        cs.rules.forEach((rule, i) => {
+          if (isEqual(rule, DEFAULT_DATE)) {
+            rulesToRemove.push(i);
+          }
+        });
+        pullAt(cs.rules, rulesToRemove);
+      });
```

Comment:
Are these blocks separated so that the variable can be named correctly (`sd`, `vs`, `cs`)? If that is the main reason, I'm wondering if we could just combine these into one block that loops over all artifacts and names the iterator something like `resource`?

Reply:
There's something that TypeScript is complaining about, which I think is due to the fact that the ValueSet and CodeSystem `rules` are not subsets of the `ExportableSdRule` type. I had separated them out because I wasn't really sure why it was upset, but I probably should take another look. So I'll look again in the morning!

Reply:
I think you can get away with this:
```
 [...this.profiles, ...this.extensions, ...this.valueSets, ...this.codeSystems].forEach(sd => {
        const rulesToRemove: number[] = [];
        sd.rules.forEach((rule: ExportableRule, i: number) => {
          if (isEqual(rule, DEFAULT_DATE)) {
            rulesToRemove.push(i);
          }
        });
        pullAt(sd.rules as ExportableRule[], rulesToRemove);
      });
```
since all rules inherit from `ExportableRule`.